client/src/js/components/bhFindDocument.js

```diff
@@ -0,0 +1,114 @@
+angular.module('bhima.components')
+.component('bhFindDocument', {
+  controller   : FindDocumentComponent,
+  controllerAs : '$ctrl',
+  templateUrl  : 'partials/templates/bhFindDocument.tmpl.html',
+  bindings: {
+    onSearchComplete     : '&',  // bind callback
+    enablePatientDetails : '<',  // bind boolean (true|false) : Enable patient details option
+    enableOptionBar      : '<',  // bind boolean (true|false) : Enable option for add, display list or display thumbnail in a bar
+    enableSearch         : '<',  // bind boolean (true|false) : Enable search bar option
+    enableSelectable     : '<',  // bind boolean (true|false) : Enable select elements
+    display              : '@',  // bind (list|thumbnail)  : Display either in list or thumbnail mode
+    patientUuid          : '<'  // Required patient uuid
+  }
+});
+
+FindDocumentComponent.$inject = [
+  'PatientService', 'ModalService', 'DocumentService',
+  'NotifyService', 'UserService', '$translate'
+];
+
+/**
+ * Find Document Component
+ * This component is responsible for displaying documents for specific patient given
+ */
+function FindDocumentComponent(Patient, Modal, Document, Notify, User, $translate) {
+  var vm = this;
+
+  /** global variables */
+  vm.session = {
+    patientUuid   : this.patientUuid,
+    patientDetail : this.enablePatientDetails,
+    display       : this.display,
+    showAction    : false
+  };
+
+  /** expose to the view */
+  vm.switchDisplay  = switchDisplay;
+  vm.toggleAction   = toggleAction;
+  vm.addDocument    = addDocument;
+  vm.deleteDocument = deleteDocument;
+  vm.mimeIcon       = mimeIcon;
+
+  // startup the component
+  startup();
+
+  /** function switchDisplay */
+  function switchDisplay(mode) {
+    vm.session.display = mode;
+  }
+
+  /** toggle document actions */
+  function toggleAction(index) {
+    vm.selectedIndex = index;
+    vm.session.showAction = vm.session.showAction === true ? false : true;
+  }
+
+  /** function add documents modal */
+  function addDocument() {
+    Modal.openUploadDocument({ patient_uuid: vm.session.patientUuid })
+    .then(startup);
+  }
+
+  /** delete document */
+  function deleteDocument(uuid) {
```

Comment:
There is a delete modal implemented in bootstrap by @lomamech.  [Here is the code](https://github.com/IMA-WorldHealth/bhima-2.X/blob/master/client/src/js/services/ModalService.js#L53).  Could you use it instead?  See #150 for the reasons why.